scandium-core/src/main/java/org/eclipse/californium/scandium/dtls/ResumingClientHandshaker.java

```diff
@@ -83,11 +117,12 @@
 	 *            if session, recordLayer or config is <code>null</code>
 	 */
 	public ResumingClientHandshaker(DTLSSession session, RecordLayer recordLayer, Connection connection,
-			DtlsConnectorConfig config, int maxTransmissionUnit) {
+			DtlsConnectorConfig config, int maxTransmissionUnit, boolean probe) {
```

Comment:
[API break] It would be easy to not break.

Reply:
Sure, but I even don't want someone encourage to create nor implement a own `Handshaker`. 

For me this is one the strange points of revapi:
A public interface may be sued for two use-cases:
- as callback interface, where it is intended to provide own implementations. Such interfaces can't be modified without major release. To add functionality requires additional interfaces (e.g. `CloseSupportingConnectionStore`) and a merge of these with the next major version.
- as "functionality expose interface", where it is not intended to provide a own implementation. Here it's very easy to add function without breaking the API.

revapi doesn't reflect this nor provide guidance to deal with. At least I didn't found such guidance. Starting issues in revapi will help out, but as I already also said, for me this "public API definition" has, amoung other tasks, so I don't want to spend more of my time in it. Feel free, to spend you own time. Maybe you start to use revapi for leshan, even if the usage there is more to gain best practice with that tool (or @sophokles73 for hono). 

As I already wrote "sometimes", without prior preparing the sources according agreed patterns and rules, it's in my opinion not too helpful, to apply strictness.
The `Handshaker`got exposed by a request of a user (checkout for the introduction of `onInitializeHandshaker`). At that time, no one concerned about unknown API design patterns. I also already asked you, to spend your time into investigate for the `Handshaker`. That is unfortunately time consuming, but would help more than insist in "not changing stuff at all". With agreed patterns and rules, the user's request at that time would have been implemented using such "expose API interfaces" and would have cause more changes, than the actually chosen approach. Fell free to spend your time in provide a proposal for that.

Please don't spend your time in answers without express your opinion on the two interface use-cases and how to setup revapi accordingly.        
     
 
That pattern

Reply:
Uups I didn't expect that you take too much time to answer to this remark. As I said in our  private discussion I didn't fight for this, I just notice it. (But maybe you didn't read this and so you could be irritated by my [API break] remarks, sorry about that)

About the 2 interface uses cases, I agree about those 2 ways to use interface.
Even if maybe, for the `expose interface` way,  using a class (facade) to do the job is maybe better. So adding method is not a problem anymore.
With java8 default implementation is a solution too.
So I would say there is many way to handle this and available solution are not the same depending of the situation. That's why I think we can not agreed about patterns or rules before to really start to do the job.

My vision when revapi raise an issue : 
 - if there is a simple way to handle it without exception we go with it
 - if there is no solution or the solution is too ugly or complex, we eventually create an exception if this make sense.
In a general way, we could also discuss with revapi developer as he probably already think about that and could be interested by our feedback.

> Maybe you start to use revapi for leshan

I will as soon as I release the 1.0.0. But we plan to consider all the public/protected stuff as part of the API. I plan 1.0.x release but I don't know if there will be a 1.1.x. The 2.0.0 will probably target LWM2M 1.1 and so will come with API break.
So I'm not sure this would help.

> I also already asked you, to spend your time into investigate for the Handshaker

I already answer about that (still in the private mail) with the interface pattern way which work but taking into account your remark about extensibility,  create a class (facade) is maybe better.

> That is unfortunately time consuming, but would help more than insist in "not changing stuff at all".

Here this is a "straw man" of my position, anyway I suppose remarks is the result of irritability generate by the fact you didn't read my last email.